### DISCUSSION TITLE
How to export private key from web3auth?

### DISCUSSION THREAD

**Discussion by 1swaraj** [Web3Auth/Web3Auth#322] [Help]
I would like to know how can one export the private key from web3auth. Also, is this dependent on the blockchain?

**Comment by BboyStatix** [comment]
Yes, exporting a private key depends on the blockchain

E.g. here's a code snippet for doing it on ethereum
```js
//Assuming user is already logged in.
async getPrivateKey() {
  const privateKey = await web3auth.provider.request({
    method: ""eth_private_key""
  });
  //Do something with privateKey
}
```
Here's one for solana
```js
async getPrivateKey() {
  const privateKey = await web3auth.provider.request({
    method: ""solanaPrivateKey""
  });
  //Do something with privateKey
}
```
And here's one for other non-evm and non-solana chains
```js
async getPrivateKey() {
  const privateKey = await web3authProvider.request({
      method: ""private_key""
  });
    //Fetch Address with privateKey
}
```

You can read more here https://web3auth.io/docs/connect-blockchain/

**Reply by timconnorz** [reply]
Is there a way for the user to do this on their own?
